partiql-eval/src/main/java/org/partiql/eval/compiler/Strategy.java

```diff
@@ -0,0 +1,41 @@
+package org.partiql.eval.compiler;
+
+import org.jetbrains.annotations.NotNull;
+import org.partiql.eval.Expr;
+import org.partiql.plan.Operator;
+
+/**
+ * Strategy converts a logical operator into a physical operator. The compiler uses the list of operands
+ * to determine a subtree match, then invokes `apply` to produce an {@link Expr}.
+ */
+public abstract class Strategy {
+
+    @NotNull
+    private final Pattern pattern;
```

Comment:
self-review: since this is an abstract class, it might be worth making the field public and removing the getter.